### DISCUSSION TITLE
Assign faces or cells after triangulation?

### DISCUSSION THREAD

**Discussion by D-Navajas** [pyvista/pyvista#2292] [Q&A] [answered]
Hi all, I need some help with my project. 

I am working with PyVista to generate several surfaces, like quadrilaterals and spheres. For quadrilaterals, the surface is made of cells with 4 points each, which I call "nodes", as they are used for thermal modelling. Here is a basic example of what I mean:

![2022-03-02_20h23_18](https://user-images.githubusercontent.com/10737095/156433893-7dd75462-0c8f-41dc-b790-03ae40f7c7c3.png)

The next step is doing ray-tracing operations with the surfaces, so I need to triangulate them. This splits each 4-point cell in half, forming two triangular cells or faces (sorry, I haven't understood the difference yet), but I would like to know which node they were part of before the triangulation. For example, with only one 4-point cell being splitted into two triangles, I would get a vector saying that both triangles were part of the first cell, something like: 
`nodes_in_model = 1`
 `faces_in_node = [0, 0] `

When it is a rectangle, is a simple operation, because each cell splits in half, but as I plan to work with removing part of the surfaces with boolean operations between surfaces, I would like to know if there is a good way to do it.

Thanks in advance

**Comment by darikg** [comment, marked as answer]
Usually it's as easy as:

```python
from pyvista import Plane
from numpy import arange

p0 = Plane()
p0.cell_data['orig_face_idx'] = arange(p0.n_cells)
p1 = p0.triangulate()
p1.cell_data['orig_face_idx']   # pyvista_ndarray([ 0, 0, 1, 1, 2, 2, ....
```

**Reply by D-Navajas** [reply]
Exactly what I needed, thanks!
